syntax = "proto3";

package chat;

// CLIENT

message UserRegistration {
    optional string username = 1;
    optional string ip = 2;
}

message UserRequest {
    optional string user = 1;
}

message ChangeStatus {
    optional string username = 1;
    optional string status = 2;
}

message MessageCommunication {
    optional string message = 1;
    optional string recipient = 2; 
    optional string sender = 3; 
}

message ClientPetition {
    optional int32 option = 1;

    optional UserRegistration registration = 2;

    optional UserRequest users = 3;

    optional ChangeStatus change = 4;

    optional MessageCommunication messageCommunication = 5;
}

// SERVER

message ConnectedUsersResponse {
    repeated UserInfo connectedUsers = 1;
}

message UserInfo {
    optional string username = 1;
    optional string status = 2;
    optional string ip = 3;
}

message ServerResponse {
    optional int32 option = 1;

    optional int32 code = 2;

    optional string serverMessage = 3;

    optional ConnectedUsersResponse connectedUsers = 4;

    optional MessageCommunication messageCommunication = 5;

    optional UserInfo userInfoResponse = 6;

    optional ChangeStatus change = 7;
}